pkg/api/api.go

```diff
@@ -245,3 +247,13 @@ func equalASCIIFold(s, t string) bool {
 	}
 	return s == t
 }
+
+type pipelineFunc func(io.Reader, int64) (swarm.Address, error)
+
+func requestPipelineFn(ctx context.Context, s storage.Storer, r *http.Request) pipelineFunc {
+	mode, encrypt := requestModePut(r), requestEncrypt(r)
+	return func(r io.Reader, l int64) (swarm.Address, error) {
```

Comment:
i. think we should redo the pipelinefunc to take the context, not the constructors